src/legacy/ui/public/filter_bar/query_filter.js

```diff
@@ -28,9 +28,19 @@ import { mapAndFlattenFilters } from './lib/map_and_flatten_filters';
 import { extractTimeFilter } from './lib/extract_time_filter';
 import { changeTimeFilter } from './lib/change_time_filter';
 
-export function FilterBarQueryFilterProvider(Promise, indexPatterns, $rootScope, getAppState, globalState, config) {
+import { getNewPlatform } from 'ui/new_platform';
+
+export function FilterBarQueryFilterProvider(Promise, indexPatterns, $rootScope, getAppState, globalState) {
   const queryFilter = {};
 
+  // ui settings handling
+  const uiSettingsClient = getNewPlatform().setup.core.uiSettings;
```

Comment:
you can move this to line 32, and you can use desctructuring:
`const { uiSettingsClient } = getNewPlatform().setup.core;`

Reply:
Figures that you can't call getNewPlatform at the root scope.
You're getting an error:

![image](https://user-images.githubusercontent.com/3016806/58238299-ac699600-7d4f-11e9-84ef-ab6fefb1a858.png)

I followed your suggetion, but inside the Provider code.